canonical/cloud-init
# cc_ntp: add support for BSDs

## Proposed Commit Message


```
cc_ntp: add support for BSDs

*BSDs have ntpd installed in base the base system
This PR extends cc_ntp to add support for ntpd,
openntpd, and chrony on the FreeBSD, and OpenBSD.

To make tests pass, we ensure that we are mocking functions,
not entire classes.

Co-authored-by: Ryan Harper <rharper@woxford.com>
Co-authored-by: James Falcon <james.falcon@canonical.com>

Sponsored by: FreeBSD Foundation

LP: #1990041
```

## Additional Context
This PR depends on #1758 

## Test Steps
- start a cloud-init vm with *BSD
- configure ntp
- observe results

## Checklist:
